metis-indexing/src/main/java/eu/europeana/indexing/IndexerPool.java

```diff
@@ -108,11 +122,12 @@ public void indexRdf(RDF stringRdfRecord, IndexingProperties indexingProperties)
   /**
    * This method removes a single record, using a free indexer in the pool
    *
-   * @param stringRdfRecord The record to be removed
+   * @param stringRdfRecord The record to be removed.
+   * @return the boolean result of the record removal.
    * @throws IndexingException In case something went wrong.
    */
-  public void remove(String stringRdfRecord) throws IndexingException {
-    indexRecord(indexer -> indexer.remove(stringRdfRecord));
+  public boolean remove(String stringRdfRecord) throws IndexingException {
+    return indexBoolTask(indexer -> indexer.remove(stringRdfRecord));
   }
 
   private void indexRecord(IndexTask indexTask) throws IndexingException {
```

Comment:
You can change this to return a value instead of duplicating the method(and also avoid creating the IndexBoolTask) since it is private.
And adapt the public methods accordingly.

Reply:
done.